### PR TITLE
Nodelet-node template utilities

### PR DESCRIPTION
**Pull request by garygra**
I've had problems debugging nodelets (need restart, crashes other nodelets, etc). In [some](https://answers.ros.org/question/35171/how-to-debug-nodelet-manager-crashes/) discussions, it is suggested to have a common implementation and launch either a node or nodelet depending on need. However, the implementation suggested that it seems convoluted and difficult to maintain.

Adding a utility for implementing nodelets so they can also be run as a simple node straightforwardly. It only requires three steps: 
  * implement nodelet as template on a base class. 
  * Instantiate nodelet using `nodelet::Nodelet`
  * Instantiate node using `utils::nodelet_as_node_t`

### CODE REVIEW

File: world/mujoco_ros/launch/mushr.launch [hunk 1]
```diff
@@ -4,26 +4,26 @@
     <arg name="visualize" default="true"/>
     <arg name="publish_ground_truth_pose" default="false"/>
     <group ns="mushr">
-        <node pkg="nodelet" type="nodelet" name="standalone_nodelet"  args="manager" output="screen"/>
+        <node pkg="nodelet" type="nodelet" name="estimation_standalone_nodelet"  args="manager" output="screen"/>
```
Comment (aravindsiv) [inline world/mujoco_ros/launch/mushr.launch]: Nit: maybe the convention we can follow is `<function>_nodelet_manager`. So this one would be `interface_nodelet_manager`.

File: estimation/launch/nodelets_as_nodes.launch
Comment (aravindsiv) [inline estimation/launch/nodelets_as_nodes.launch:1]: Can we have this `.launch` file use the same `roslaunch` params as the nodelet's `.launch` file? That way, when we are debugging, we change only a single file to reproduce.

File: infrastructure/utils/launch/nodelet_debug.launch
Comment (aravindsiv) [inline infrastructure/utils/launch/nodelet_debug.launch:1]: Why is this file here? Isn't this logic part of a larger `roslaunch` file?

**Review by aravindsiv — CHANGES_REQUESTED** [review]
Thx for the PR - just need some clarifications before we can merge.

File: estimation/launch/nodelets_as_nodes.launch
Comment (garygra) [reply estimation/launch/nodelets_as_nodes.launch:1]: Removing this file to have only the debug one.

File: infrastructure/utils/launch/nodelet_debug.launch
Comment (garygra) [reply infrastructure/utils/launch/nodelet_debug.launch:1]: I used for debugging, removing it.

File: estimation/launch/nodelets_as_nodes.launch
Comment (aravindsiv) [reply estimation/launch/nodelets_as_nodes.launch:1]: We should keep one copy of this logic in a bigger `.launch` file, or move its contents to `nodelets_debug.launch`.

**Review by aravindsiv — CHANGES_REQUESTED** [review]
Minor comments to restore the nodelets debug example.

**Review by aravindsiv — APPROVED** [review]
LGTM, merging.

**Pull request merged**